src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -76,6 +78,101 @@ def chunk_iter(inputs, feature_extractor, chunk_len, stride_left, stride_right):
             yield {"is_last": is_last, "stride": stride, **processed}
 
 
+def _find_timestamp_sequence(sequences, tokenizer, feature_extractor):
+    """ """
+    max_source_positions = 1500
+    begin_idx = 3
+    last_begin_time = 0
+    items = []
+    offsets = []
+    # index of the first timestamp token
+    timestamp_begin = tokenizer.all_special_ids[-1] + 1
+    time_precision = feature_extractor.chunk_length / max_source_positions
+    last_timestamp_position = 0
+    for seq_idx, sequence in enumerate(sequences):
+        timestamp_offset = (seq_idx * feature_extractor.chunk_length) / 2  # in seconds
+        sequence = sequence.squeeze(0)
+        # patch the matching sequence with the correct sequence of the previous chunk
+        sequence = sequence[begin_idx:]
+        if last_timestamp_position != 0:
+            # lets only take the beginning tokens
+            timestamp_tokens = np.where(sequence >= timestamp_begin)[0][1::2]
+            time = (sequence[timestamp_tokens] - timestamp_begin) * 0.02 + timestamp_offset
+            merge_point = np.where(time > last_begin_time)[0][0] - 1
+            offset_idx = -1
+            for idx, item in enumerate(offsets):
+                if item["start_time"] >= time[merge_point]:
+                    offset_idx = idx - 1
+                    break
+            sliced_sequence = sequence[timestamp_tokens[merge_point] + 2 : timestamp_tokens[merge_point + 1] + 1]
+            prev_sequences = items[offset_idx][:-1]
+            for prev_sequence in items[offset_idx + 1 :]:
+                prev_sequences = np.concatenate([prev_sequences, prev_sequence[1:-1]])
+            start_timestamp_position = offsets[offset_idx]["start_time"]
+
+            items = items[:offset_idx]
+            offsets = offsets[:offset_idx]
+            # here sometime you can have repetition if prev_sequences[None,:] exactly in
+            patch = _find_longest_common_sequence([prev_sequences[None, :], sliced_sequence[None, :]], tokenizer)
```

Comment:
Seems odd you need to resort to this.

If you have whisper's timestamps, they should be the source of truth, not this.

Reply:
Problem is sometimes the sentences are not split the same way, and thus you don't have matching timestamps